test/fit_tests/common/fit_common.py

```diff
@@ -0,0 +1,528 @@
+    "obmmac": os.getenv("OBMMAC", "all"), # node OBM MAC address
+    "nodeid": os.getenv("NODEID", "None"), # node ID
+    "hyper": "None", # hypervisor address
+    "version": os.getenv("VERSION", "onrack-devel"), # code version
+    "xunit": os.getenv("XUNIT", False), # XUNIT output
+    "list": os.getenv("LIST", False), # list tests
+    "group": os.getenv("GROUP", "all"), # test group
+    "http": os.getenv("HTTP", "False"), # force http api protocol
+    "https": os.getenv("HTTPS", "False"), # force https api protocol
+}
+
+# Get top level path via git
+TEST_PATH = subprocess.check_output("git rev-parse --show-toplevel", shell=True).rstrip("\n") + "/test/fit_tests/"
+CONFIG_PATH = TEST_PATH + ARGS_LIST["config"] + "/"
+if ARGS_LIST["config"] != 'config':
+    print "**** Using config file path:", ARGS_LIST["config"]
+VERBOSITY = int(os.getenv("VERBOSITY", "1"))
+GLOBAL_CONFIG = []
+STACK_CONFIG = []
+API_PORT=None
+API_PROTOCOL=None
+
+# List of BMC IP addresses
+BMC_LIST = []
+
+
+# Global Config files
+try:
+    GLOBAL_CONFIG = json.loads(open(CONFIG_PATH + "global_config.json").read())
+except:
+    print "**** Global Config file: " + CONFIG_PATH + "global_config.json" + " missing or corrupted! Exiting...."
+    sys.exit(255)
+try:
+    STACK_CONFIG = json.loads(open(CONFIG_PATH + "stack_config.json").read())
+except:
+    print "**** Stack Config file:" + CONFIG_PATH + "stack_config.json" + " missing or corrupted! Creating empty stack file...."
+    STACK_CONFIG = []
+
+# apply stack detail files from config dir to STACK_CONFIG dict
+for entry in os.listdir(CONFIG_PATH):
+    if entry != "global_config.json" and entry != "stack_config.json" and ".json" in entry:
+        try:
+            detailfile = json.loads(open(CONFIG_PATH + entry).read())
+        except:
+            print "**** Invalid JSON file:", CONFIG_PATH + entry
+        else:
+            STACK_CONFIG.update(detailfile)
+
+
+# This section derives default stack configuration data from STACK-CONFIG, use environment to override
+ARGS_LIST.update(
+    {
+        "usr": GLOBAL_CONFIG['credentials']['ora'][0]['username'],
+        "pwd": GLOBAL_CONFIG['credentials']['ora'][0]['password']
+    }
+)
+
+if ARGS_LIST["stack"] != "None":
+    if "ora" in STACK_CONFIG[ARGS_LIST["stack"]]:
+        ARGS_LIST["ora"] = STACK_CONFIG[ARGS_LIST["stack"]]['ora']
+    else:
+        ARGS_LIST["ora"] = "localhost"
+    if "bmc" in STACK_CONFIG[ARGS_LIST["stack"]]:
+        ARGS_LIST["bmc"] = STACK_CONFIG[ARGS_LIST["stack"]]['bmc']
+    if "hyper" in STACK_CONFIG[ARGS_LIST["stack"]]:
+        ARGS_LIST["hyper"] = STACK_CONFIG[ARGS_LIST["stack"]]['hyper']
+
+# set api port and protocol
+if ARGS_LIST['http'] == "True":
+    API_PROTOCOL = 'http'
+    API_PORT = GLOBAL_CONFIG['ports']['http']
+if ARGS_LIST['https'] == "True":
+    API_PROTOCOL = 'https'
+    API_PORT = GLOBAL_CONFIG['ports']['http']
+if ARGS_LIST["ora"] == "localhost":
+    API_PROTOCOL = 'http'
+    API_PORT = '8080'
+
```

Comment:
Better to add some assertion logic here to check whether some required arguments are missing and whether there are illegal values so that following code will not fall into the trouble for argument validation.